autogen/header.rs

```diff
@@ -100,27 +100,29 @@ fn gen_value_enum_operand_kind(grammar: &structs::OperandKind) -> TokenStream {
     let mut from_prim_list = vec![];
     let mut aliases = vec![];
     let mut capability_clauses = BTreeMap::new();
+    let mut from_str_impl = vec![];
     for e in &grammar.enumerants {
         if let Some(discriminator) = seen_discriminator.get(&e.value) {
             let symbol = as_ident(&e.symbol);
             aliases.push(quote! {
-                pub const #symbol: #kind = #kind::#discriminator;
+                pub const #symbol: Self = Self::#discriminator;
             });
         } else {
             // Special case for Dim. Its enumerants can start with a digit.
             // So prefix with the kind name here.
-            let name = if grammar.kind == "Dim" {
+            let name_str = if grammar.kind == "Dim" {
                 let mut name = "Dim".to_string();
                 name.push_str(&e.symbol);
                 name
             } else {
                 e.symbol.to_string()
             };
-            let name = as_ident(&name);
+            let name = as_ident(&name_str);
             let number = e.value;
             seen_discriminator.insert(e.value, name.clone());
             enumerants.push(quote! { #name = #number });
-            from_prim_list.push(quote! { #number => #kind::#name });
+            from_prim_list.push(quote! { #number => Self::#name });
+            from_str_impl.push(quote! { #name_str => Ok(Self::#name), });
```

Comment:
I think the aliases should probably also appear. Right now `ClosestHitKHR` isn't recognised but `ClosestHitNV` is for example, however the new way forward it through the KHR extension.